Source/WebCore/dom/SimulatedClick.cpp

```diff
@@ -77,11 +79,55 @@ class SimulatedMouseEvent final : public MouseEvent {
     }
 };
 
+class SimulatedPointerEvent final : public PointerEvent {
+    WTF_MAKE_TZONE_OR_ISO_ALLOCATED_INLINE(SimulatedPointerEvent);
+public:
+    static RefPtr<SimulatedPointerEvent> create(const AtomString& type, const SimulatedMouseEvent& event, RefPtr<Event>&& underlyingEvent, Element& target, SimulatedClickSource source)
+    {
+        return adoptRef(*new SimulatedPointerEvent(type, event, WTFMove(underlyingEvent), target, source));
+    }
+
+private:
+    // If the device type cannot be detected by the user agent, then the value MUST be an empty string.
+    static constexpr auto pointerType = ASCIILiteral { ""_s };
+
+    // The pointerId value of -1 MUST be reserved and used to indicate events that were generated by something other than a pointing device.
+    static constexpr auto pointerID = static_cast<PointerID>(-1);
```

Comment:
We should fix this separately, but this will be wrong. Currently this ID will suffer from integer underflow. The spec says `pointerID` should be a `long` value, i.e. `int32_t`. However, in [263829@main](https://commits.webkit.org/263829@main), I naively changed that to be a `uint32_t` instead to deal with ID overflows. Back then I hadn't realized that negative IDs were valid. We should consider bumping the ID type to be an `int64_t`, instead.

Reply:
hm, there is explicitly a WPT test that checks if it's -1, and it passes, somehow

Reply:
Interesting!

Reply:
Do you also locally log a -1 ID for these events?